backend/db/utils.py

```diff
@@ -177,21 +179,23 @@ def refresh_termhub_core_cset_derived_tables(con: Connection, schema=SCHEMA, pol
         i += 1
         if (datetime.now() - t0).total_seconds() >= 2 * 60 * 60:  # 2 hours
             raise RuntimeError('Timed out after waiting 2 hours for other active derived table refresh to complete.')
-        elif check_db_status_var('derived_tables_refresh_status') == 'active':
+        # As of 2024/01/21 this is now a redundancy. Concerning the main refresh (refresh.py), it won't even begin if
+        # ...the derived refresh is active.
+        elif is_derived_refresh_active(local):
             msg = f'Another derived table refresh is active. Waiting {polling_interval_seconds} seconds to try again.' \
                 if i == 0 else '- trying again'
             print(msg)
             time.sleep(polling_interval_seconds)
         else:
             try:
-                update_db_status_var('derived_tables_refresh_status', 'active')
+                update_db_status_var('last_derived_refresh_request', current_datetime(), local)
                 # The following two calls yield equivalent results as of 2023/08/08. I've commented out
                 #  refresh_derived_tables() in case anything goes wrong with refresh_any_dependent_tables(), since that
                 #  is based on a heuristic currently, and if anything goes wrong, we may want to switch back. -joeflack4
                 # refresh_derived_tables(con, CORE_CSET_DEPENDENT_TABLES, schema)
                 refresh_any_dependent_tables(con, CORE_CSET_TABLES, schema)
             finally:
-                update_db_status_var('derived_tables_refresh_status', 'inactive')
```

Comment:
## Replaced `inactive`/`active` with start/stop time
Now the derived table refresh activity monitoring is done the same way as the full database refresh. 

This fixes the problem where the refresh exits unexpectedly and the except/finally blocks do not execute, e.g. when cancelling a run in development environment, or when GH action times out.